finality-aleph/src/nodes/validator_node.rs

```diff
@@ -42,10 +48,40 @@ where
         millisecs_per_block,
         justification_rx,
         backup_saving_path,
+        external_addresses,
+        validator_port,
         ..
     } = aleph_config;
 
-    let (validator_network, validator_network_identity) = tcp_network::new_noop().await;
+    // We generate the phrase manually to only save the key in RAM.
```

Comment:
how will this work in case a node crashes and rejoin the network during the session?

Reply:
A new authentication will be generated and broadcast. These keys changing was already taken into account when writing validator node connection management, because it technically could have happened with libp2p peer addresses anyway, now it's only more likely.